extras/create_send_notifications.py

```diff
@@ -42,26 +42,49 @@ def create_output_directories():
-
-    This is the list of CyHy organization IDs (and their descendants) that
-    receive CyHy reports.
+    This is the list of organization IDs that should
+    have a notification generated and sent.
+    """
+    notifications_to_generate = set()
+    cyhy_parent_ids = set()
+    ticket_owner_ids = db.NotificationDoc.collection.distinct("ticket_owner")
+    for request in db.RequestDoc.collection.find({"_id": {"$in": ticket_owner_ids}, "report_types": "CYHY"}, {"_id":1}):
+        # If the notification document's ticket owner has "CYHY" in their list of report_types,
+        # then a notification should be generated for that owner:
+        notifications_to_generate.add(request["_id"])
+        logging.debug("Added {} to notifications_to_generate".format(request["_id"]))
+        # Recursively check for any ancestors of the ticket owner that have "CYHY" in
+        # their list of report_types.  If found, add them to the list of owners that
+        # should get a notification.
+        cyhy_parent_ids.update(find_cyhy_parents(db, request["_id"]))
+    notifications_to_generate.update(cyhy_parent_ids)
+    notifications_to_delete = set(ticket_owner_ids) - notifications_to_generate
+    return list(notifications_to_generate), list(notifications_to_delete)
+          
+def find_cyhy_parents(db, org_id):
+    """Return parents/grandparents/etc. of an organization that have "CYHY" in their list of report_types.
     """
-    cyhy_org_ids = set()  # Use a set here to avoid duplicates
-    for cyhy_request in list(
-        db.RequestDoc.collection.find(
-            {"report_types": "CYHY"}, {"_id": 1, "children": 1}
-        ).sort([("_id", 1)])
-    ):
-        cyhy_org_ids.add(cyhy_request["_id"])
-        if cyhy_request.get("children"):
-            cyhy_org_ids.update(db.RequestDoc.get_all_descendants(cyhy_request["_id"]))
-    return list(cyhy_org_ids)
-
-
-def generate_notification_pdfs(db, org_ids, master_report_key):
+    cyhy_parents = set()
+    for request in db.RequestDoc.collection.find({"children": org_id}, {"_id": 1, "report_types": 1}):
+        if "CYHY" in request["report_types"]:
+            # There is an undocumented constraint where organizations are set up
+            # with only one level of children (i.e. no grandchildren orgs). Given 
+            # the fact that it is undocumented, following the hierarchy to the top
+            # is the best solution. We do not expect to go up more than one level,
+            # but we check all ancestors just to be sure.
+            cyhy_parents.add(request["_id"])
+            # Found a parent of org_id with "CYHY" in their list of report_types,
+            # so add it to our set
+            logging.debug("{} - Adding to set of CYHY parents".format(request["_id"]))
+        # Recursively call find_cyhy_parents() to check if this org has any parents
+        # with "CYHY" in their list of report_types
+         cyhy_parent_ids.update(find_cyhy_parents(db, request["_id"]))
```

Comment:
An extra space snuck in here somehow:
```suggestion
        cyhy_parent_ids.update(find_cyhy_parents(db, request["_id"]))
```

Reply:
Fixed in https://github.com/cisagov/cyhy-reports/pull/96/commits/9e4dcc1ae51e83a30b2a14d084f25069c9acf37e